libsigner/src/messages.rs

```diff
@@ -383,6 +402,10 @@ impl StacksMessageCodec for SignerMessage {
                     party_polynomials,
                 }
             }
+            SignerMessageTypePrefix::EncryptedSignerState => {
+                let encrypted_state = read_next::<_, _>(fd)?;
```

Comment:
There needs to be a `BoundReader` wrapping `fd` so as to impose a maximum amount of bytes that will be read from it.  Do we know how big this blob of data can be?  Can it fit in a StackerDB slot?

Reply:
Polynomial size is threshold*32 bytes, key shares are |shares|*32 bytes.  So worst case scenario is like 32MB, for 4k total reward slots and 250 owned key_ids.

Is there some hard max size for a slot, or a performed cliff?

Reply:
Yikes, that's too big.  Network messages max out at 2MB.

Reply:
Hmm, this may require some re-thinking.  Unfortunately, we can't just increase the maximum network message size, because that's the only limit stopping blocks from getting too big (meaning, it'd be highly disruptive to change this without some kind of gating).  Splitting the state into multiple chunks isn't too appealing either, because then there needs to be extra logic for that (and there'd need to be 16 more StackerDBs just for holding each signer's state).

Reply:
LoL sorry that should be `(2800+250)*32` bytes, which is much more reasonable.

Reply:
Oh great, that's only ~180k

Reply:
Great. Note that this will be extended in https://github.com/stacks-network/stacks-core/issues/4654 though, but we can probably limit it to a factor two so we should still have a good margin. I'll make a note in that ticket.

I'll update to use a `BoundReader`.

Reply:
Done. I added a factor 4 to the bound to have margin for the other fields in the saved state beyond the dkg shares. Let me know if you prefer to have the bound be more exact or if you want another margin.

So the bound right now is (2800 + 2800) * 32 * 4.

Reply:
Resolving this conversation. Any suggestions on alternative sizes of the margin should be in this thread: https://github.com/stacks-network/stacks-core/pull/4659#discussion_r1563121809